packages/core/database/lib/dialects/mysql/schema-inspector.js

```diff
@@ -161,7 +161,11 @@ class MysqlSchemaInspector {
         ret[index.Key_name] = {
           columns: [index.Column_name],
           name: index.Key_name,
-          type: !index.Non_unique ? 'unique' : null,
+          type: !index.Non_unique
+            ? 'unique'
+            : index.Index_type === 'FULLTEXT'
```

Comment:
Is fulltext the only other option ? 
Feels like we could just lowercase the index_type and use it as is in knex 🤔  wdyt ?

Reply:
Cannot. `Index_type` is enum (`BTREE`, `FULLTEXT`).
according to the output of mysql `show index from table_name`

Reply:
so we should handle BTREE here too no? 🤔

Reply:
I think we can implement the type `SPATIAL` instead of the implementation types B-Tree or R-Tree or Hash. As knex does not handle the [`index_option`](https://dev.mysql.com/doc/refman/8.0/en/create-index.html) options when generating the SQL. The knex query only handles the index type after the `CREATE` keyword (i.e. `CREATE ${type} INDEX index_name`)

Reply:
It is possible with a param called storageEngineIndexType.  (ref: http://knexjs.org/guide/schema-builder.html#index) but we can address that later. 

It would be worth to support the spatial one then and in a later iteration we can add the storageEngine option